src/lib/encoder.ts

```diff
@@ -143,6 +143,123 @@ const decodeCompactBytesArray = (compactBytesArray: string): string[] => {
   return encodedValues;
 };
 
+const encodeBytesNCompactBytesArray = (
+  values: string[],
+  numberOfBytes: number,
+): string => {
+  for (let i = 0; i < values.length; i++) {
+    if (values[i].length > numberOfBytes * 2 + 2)
+      throw new Error(
+        `Hex bytes${numberOfBytes} value at index: ${i} is using ${
+          (values[i].length - 2) / 2
+        } bytes, which exceedes ${numberOfBytes}`,
+      );
+  }
+
+  return encodeCompactBytesArray(values);
+};
+
+const decodeBytesNCompactBytesArray = (
+  compactBytesArray: string,
+  numberOfBytes: number,
+): string[] => {
+  const bytesValues = decodeCompactBytesArray(compactBytesArray);
+
+  for (let i = 0; i < bytesValues.length; i++) {
+    if (bytesValues[i].length > numberOfBytes * 2 + 2)
+      throw new Error(
+        `Hex bytes${numberOfBytes} value at index: ${i} is using ${
+          (bytesValues[i].length - 2) / 2
+        } bytes, which exceedes ${numberOfBytes}`,
+      );
+  }
+
+  return bytesValues;
+};
+
+const returnCompactBytesArrayBytesNTypes = () => {
```

Comment:
The name of the function is not very clear to what it is doing ?

Reply:
yep, sorry, it just returns:
{
    'uint8[CompactBytesArray]' = {
        encode: (value: number[]) => encodeUintNCompactBytesArray(value, 1),
        decode: (value: string) => decodeUintNCompactBytesArray(value, 1),
},
{
    'uint16[CompactBytesArray]' = {
        encode: (value: number[]) => encodeUintNCompactBytesArray(value, 2),
        decode: (value: string) => decodeUintNCompactBytesArray(value, 2),
},
[...]
{
    'uint256[CompactBytesArray]' = {
        encode: (value: number[]) => encodeUintNCompactBytesArray(value, 32),
        decode: (value: string) => decodeUintNCompactBytesArray(value, 32),
}

Reply:
@magalimorin18 @b00ste can we resolve this comment?

Reply:
I changed the function name and added JSdocs, I would say yes on my end, wdyt @magalimorin18 ?